dev/benchmarks/macrobenchmarks/test/frame_policy.dart

```diff
@@ -0,0 +1,105 @@
+
+void main() {
+  final E2EWidgetsFlutterBinding binding =
+      E2EWidgetsFlutterBinding.ensureInitialized() as E2EWidgetsFlutterBinding;
+  testWidgets(
+    'Frame Counter and Input Delay for benchmarkLive',
+    (WidgetTester tester) async {
+      await tester.pumpWidget(MaterialApp(home: Scaffold(body: SimpleScroll())));
+      await tester.pumpAndSettle();
+      final Offset location = tester.getCenter(find.byType(ListView));
+      int frameCount = 0;
+      final FrameCallback frameCounter = (Duration elapsed) {
+        frameCount += 1;
+      };
+      tester.binding.addPersistentFrameCallback(frameCounter);
+
+      const int timeInSecond = 1;
+      const Duration totalTime = Duration(seconds: timeInSecond);
+      const int moveEventNumber = timeInSecond * 120;  // 120Hz
+      const Offset movePerRun = Offset(0.0, -200.0 / moveEventNumber);
+      final List<PointerEventRecord> records = <PointerEventRecord>[
+        PointerEventRecord(Duration.zero, <PointerEvent>[
+          PointerAddedEvent(
+            timeStamp: Duration.zero,
+            position: location,
+          ),
+          PointerDownEvent(
+            timeStamp: Duration.zero,
+            position: location,
+            pointer: 1,
+          ),
+        ]),
+        ...<PointerEventRecord>[
+          for (int t=0; t < moveEventNumber; t++)
+            PointerEventRecord(totalTime * (t / moveEventNumber), <PointerEvent>[
+              PointerMoveEvent(
+                timeStamp: totalTime * (t / moveEventNumber),
+                position: location + movePerRun * t.toDouble(),
+                pointer: 1,
+                delta: movePerRun,
+              )
+            ])
+        ],
+        PointerEventRecord(totalTime, <PointerEvent>[
+          PointerUpEvent(
+            // Deviate a little from integer number of frames to reduce flakiness
+            timeStamp: totalTime - const Duration(milliseconds: 1),
+            position: location + movePerRun * moveEventNumber.toDouble(),
+            pointer: 1,
+          )
+        ])
+      ];
+
+      binding.framePolicy = LiveTestWidgetsFlutterBindingFramePolicy.benchmarkLive;
+      List<Duration> delays = await tester.handlePointerEventRecord(records);
+      await tester.pumpAndSettle();
+      _reportResult('Test Event Delay:benchmarkLive', frameCount, delays);
+      await tester.idle();
+      await tester.binding.delayed(const Duration(milliseconds: 250));
+
+      binding.framePolicy = LiveTestWidgetsFlutterBindingFramePolicy.fullyLive;
+      frameCount = 0;
+      delays = await tester.handlePointerEventRecord(records);
+      await tester.pumpAndSettle();
+      _reportResult('Test Event Delay:fullyLive', frameCount, delays);
+      await tester.idle();
+    },
+  );
+}
+
+void _reportResult(String eventName, int frameCount, List<Duration> delays) {
+  assert(delays.length > 1);
+  final List<int> delayedInMicro = delays.map<int>(
+    (Duration delay) => -delay.inMicroseconds,
```

Comment:
This looks strange. Does `handlePointerEventRecord` mostly return all negative delays? Shall we use `abs` just in case of positive delays? We might need to add some documentations to `handlePointerEventRecord` to describe the meaning of positive and negative delays, and set the expectation if most of them are negative or positive.

Reply:
Yes mostly negative. It's positive only when `Future.delayed` complete earlier than the set time. My local test shows it's 0~3 out of 200+ events with tiny positive values < 100us. I don't think I should use `abs` because this list is only shown in debug mode and `abs` makes less use of the info. But it does make sense to change the behavior of `handlePointerEventRecord` for the major result being positive since we are calling it delay, and add docs to that.

Reply:
Update about the number. When putting this on a more complex environment, the fluctuation becomes larger, and positives can go as much as ~10% and up to 500us. Still tiny compare to frame interval though.